src/game.cpp

```diff
@@ -11602,14 +11602,54 @@ void game::read()
     auto filter = []( const item &it ) {
         return it.is_book();
     };
-    int pos = inv_for_filter( _("Read:"), filter );
 
-    if (pos == INT_MIN) {
-        add_msg(_("Never mind."));
+    auto item_loc = inv_map_splice( filter, _("Read:") );
+
+    const int inv_pos = item_loc.get_inventory_position();
+    item *it = item_loc.get_item();
+
+    if ( it == nullptr ) {
+        add_msg( _("Never mind.") );
         return;
     }
+
+    /* although working properly currently without the below, but just put in as a safeguard to future changes
+       where inv_map_splice return a non book item
```

Comment:
FYI: The future is now! The inventory function may return any item of the inventory. The filter will only prevent the items from being displayed, but the player can still select them directly with the inventory letter.

Thanks BevapDin

You are right. Currently already returning worn items (including weapons)... The comment is wrong... shall take it out